setup.py

```diff
@@ -141,6 +142,7 @@
         'pyparsing',
         'scipy',
         'six',
+        'bokeh'
```

Comment:
I wonder if we should add a new `visualization` section under `optional_dependencies`... 
I don't think we want this as a core dependency? (would have to add appropriate handling of failed import)

Reply:
Yeah, I think that's reasonable to create a new section of optional dependencies